tests/llm/clients/test_client_config.py

```
@@ -247,6 +247,207 @@ async def test_creates_nim(self):
+            assert model._client._custom_headers == {"X-Custom": "value", "api-key": "k"}
+            assert model._client._custom_query == {"trace-id": "abc", "api-version": "2024-02-15-preview"}
+        finally:
+            await fw.reset()
+
+    @pytest.mark.asyncio
+    async def test_azure_missing_endpoint_raises(self):
+        from nemoguardrails.llm.frameworks.default import DefaultFramework
+
+        fw = DefaultFramework()
+        try:
+            with pytest.raises(ValueError, match="requires parameters.azure_endpoint"):
```

Ruff (0.15.12)

[warning] Pattern passed to `match=` contains metacharacters but is neither escaped nor raw

(RUF043)